htlcswitch/interceptable_switch.go

```diff
@@ -136,16 +143,28 @@ type InterceptableSwitchConfig struct {
 	Notifier chainntnfs.ChainNotifier
 
 	// CltvRejectDelta defines the number of blocks before the expiry of the
-	// htlc where we no longer intercept it and instead cancel it back.
+	// htlc where we auto-fail an intercepted htlc to prevent channel
+	// force-closure.
 	CltvRejectDelta uint32
 
+	// CltvInterceptDelta defines the number of blocks before the expiry of
+	// the htlc where we don't intercept anymore. This value must be greater
+	// than CltvRejectDelta, because we don't want to offer htlcs to the
+	// interceptor client for which there is no time left to resolve them
+	// anymore.
+	CltvInterceptDelta uint32
+
 	// RequireInterceptor indicates whether processing should block if no
 	// interceptor is connected.
 	RequireInterceptor bool
 }
 
 // NewInterceptableSwitch returns an instance of InterceptableSwitch.
 func NewInterceptableSwitch(cfg *InterceptableSwitchConfig) *InterceptableSwitch {
+	if cfg.CltvInterceptDelta <= cfg.CltvRejectDelta {
+		panic("cltv intercept delta not greater than cltv reject delta")
```

Comment:
This is hacky. Can we return an error here or do some kind of validation of the config?